src/dc_receive_imf.rs

```diff
@@ -266,12 +245,9 @@ pub fn dc_receive_imf(
         "received message {} has Message-Id: {}", server_uid, rfc724_mid
     );
 
-    cleanup(
-        context,
-        &create_event_to_send,
-        &created_db_entries,
-        &rr_event_to_send,
-    );
+    cleanup(context, &create_event_to_send, &created_db_entries);
+
+    mime_parser.handle_reports(from_id, sent_timestamp, &server_folder, server_uid);
```

Comment:
Why move `handle_reports` call?

Reply:
The actual events got sent a cleanup() time previously.  I wasn't sure if there were considerations behind this "send events at the end" and just kept the sending close to where it happened before. pretty sure that nothing modifies self.reports between the old place and the new.